### PR TITLE
Add launch.Launchable interface

### PR DESCRIPTION
**Pull request by bwester**
Extracts `hoist.Launchable` methods into an interface `launch.Launchable` in
preparation for creating new types of launchables.

The interface is placed into its own package to avoid circular or unintuitive
dependencies. The package name "launch" was chosen to avoid having to rewrite
every "launchable" identifier used as a local variable.

It explicitly does *not* attempt to improve upon the basic set of methods
exposed by `hoist.Launchable`, though with much further thought and another
implementation, there may be ample room for improvement.

### CODE REVIEW

**Comment by mpuncel** [comment]
:+1:

File: pkg/pods/pod.go [hunk 1]
```diff
@@ -100,12 +101,12 @@ func (pod *Pod) Halt(manifest Manifest) (bool, error) {
...
-		case hoist.DisableError:
+		case launch.DisableError:
 			// do not set success to false on a disable error
-			pod.logLaunchableWarning(launchable.Id, err, "Could not disable launchable")
+			pod.logLaunchableWarning(launchable.GetID(), err, "Could not disable launchable")
 		default:
-			// this case intentionally includes hoist.StopError
-			pod.logLaunchableError(launchable.Id, err, "Could not halt launchable")
+			// this case intentionally includes launch.StopError
+			pod.logLaunchableError(launchable.GetID(), err, "Could not halt launchable")
```
Comment (anthonybishopric) [inline pkg/pods/pod.go]: Consider `ID()` to be consistent with `Pod` and the other getters on the struct?

**Comment by anthonybishopric** [comment]
:+1: except for nit

File: pkg/pods/pod.go (hunk 1 above)
Comment (bwester) [inline pkg/pods/pod.go]: There's a naming conflict between `host.Launchable` having an `Id` field and an `ID()` method.  It's not a strict conflict in this case, but it's too close for my comfort.

But you reminded me that we do have a convention of interfaces with `ID()`, so I'm more inclined now to rename the field.

**Comment by petertseng** [comment]
On an unrelated note, it may be appropriate for `fields.RC` 's `Id()` method to be renamed to `ID()` ... separate PR of course

**Comment by tummychow** [comment]
> On an unrelated note, it may be appropriate for fields.RC 's Id() method to be renamed to ID() ... separate PR of course

I'll take care of it in refactors part 2

File: pkg/pods/pod.go (hunk 1 above)
Comment (bwester) [inline pkg/pods/pod.go]: Naming getters without the "Get" is a an explicit [language convention](https://golang.org/doc/effective_go.html#Getters), but it's less clear what to do when you really want a public field with that name too. Any strong opinions on naming a public field `ID_` and its getter `ID()`?

Comment (bwester) [inline pkg/pods/pod.go]: Never mind.  I should probably just do the right thing and not have any of these fields be exported.

**Pull request merged**